release-tools/debian-package-builder

```diff
@@ -0,0 +1,106 @@
+#   "debian" branch, inside the directory ./debian/patches. Make sure each
+#   patch is listed in the file ./debian/patches/series. Then re-run
+#   $ dpkg-buildpackage -S
+#   The resulting package will be placed in the parent directory of the root of
+#   the project, e.g. in $HOME/packaging/src/github.com/snapcore, assuming a
+#   setup where the GOPATH was set to $HOME/packaging
+# - build the binary package
+#   While being in the directory $HOME/packaging/src/github.com/snapcore issue
+#   ./debian-packaging-builder build-on-debian-10
+#   This step will copy the source package to an appropriate multipass virtual
+#   machine and build it there using sbuild. Any build dependencies obtained in
+#   this step are automatically cached and are quick to reuse on subsequent
+#   runs.
+#
+#   In case the build fails an interactive shell will be spawned by sbuild, still
+#   running inside the virtual machine. The package is unpacked twice, you
+#   really only care about the files in the _build directory. You can set
+#   GOPATH manually to
+#   GOPATH=/path/to/directory/called/_build:/usr/share/gocode and run tests and
+#   iterate as usual. The source tree alongside the _build directory is useful
+#   for comparison and for generating patches..
+
+# There must be an apt-cacher-ng installation on the host.
+proxy_url=http://$(shell ip -br address show dev eth0 | awk '{print $$3}' | cut -d / -f 1):3142
+
+# Packages that are not required for building snapd but are useful for interactive development.
+packages=git sbuild mc devscripts avahi-daemon ssh emacs-nox vim-nox rsync
+
+# Helpful function for running a command on the remote machine.
+# Assumes that the target name ends with machine name, like below.
+remote=multipass exec $* --
+
+PHONY: help
+help:
+	@echo "Availale targets"
+	@echo "  create-NAME: create a multipass virtual machine"
+	@echo "  prepare-NAME: prepare a virtual machine for building packages"
+	@echo "  destroy-NAME: destroy a multipass virtual machine"
+	@echo "  build-on-NAME: copy a source package and build it on a virtual machine"
+	@echo
+	@echo "Available names are: debian-9 debian-10"
+	@echo "NOTE: You must provide the source package yourself"
+	@echo "Please read the script for additional instructions."
+
+.PHONY: create-debian-9
+create-debian-9:
+	multipass launch -n debian-9 -c 8 -m 4G https://cdimage.debian.org/mirror/cdimage/openstack/current/debian-9.6.4-20190118-openstack-amd64.qcow2
+
+.PHONY: create-debian-10
+create-debian-10:
+	multipass launch -n debian-10 -c 8 -m 4G https://cdimage.debian.org/mirror/cdimage/openstack/testing/debian-testing-openstack-amd64.qcow2
+
+.PHONY: prepare-debian-9 prepare-debian-10
+prepare-debian-9 prepare-debian-10: prepare-%:
+	$(remote) sudo mkdir -p /etc/apt/apt.conf.d/
+	$(remote) sudo sh -c 'echo "Acquire::HTTP::Proxy \"$(proxy_url)\";" > /etc/apt/apt.conf.d/00proxy'
+	-$(remote) sudo apt-get update
```

Comment:
NOTE: this fails on sid, because of missing Release file in one of the repositories.